include/device.hpp

```diff
@@ -0,0 +1,574 @@
+
+struct DI_JB {
+  quint64 lba;
+  float spdx;
+  float jitter;
+  float asymm;
+};
+
+struct MM_JB {
+  float jmin;
+  float jmax;
+  float bmin;
+  float bmax;
+};
+
+struct DI_FT {
+  quint64 lba;
+  float spdx;
+  quint32 fe;
+  quint32 te;
+};
+
+struct DI_TA {
+  // float T;
+  int idx;
+  int pit;
+  int land;
+};
+
+class TestData {
+public:
+  TestData(){};
+  ~TestData(){};
+  void clear() {
+    clearRT();
+    clearWT();
+    clearErrc();
+    clearJB();
+    clearFT();
+    clearTA();
+  };
+
+  inline void clearRT() {
+    rt.clear();
+    rt_time = 0;
+  };
+  inline void clearWT() {
+    wt.clear();
+    wt_time = 0;
+  };
+  inline void clearErrc() {
+    errc.clear();
+    errc_time = 0;
+
+    errcTOT.raw.spdx = 0.0;
+    for (int i = 0; i < 8; i++)
+      errcTOT.raw.err[i] = 0;
+    errcMAX.raw.spdx = 0.0;
+    for (int i = 0; i < 8; i++)
+      errcMAX.raw.err[i] = 0;
+    errcAVG.raw.spdx = 0.0;
+    for (int i = 0; i < 8; i++)
+      errcAVG.raw.err[i] = 0.0;
+  };
+  inline void clearJB() {
+    jb.clear();
+    jb_time = 0;
+
+    jbMM.jmin = 0.0;
+    jbMM.jmax = 0.0;
+    jbMM.bmin = 0.0;
+    jbMM.bmax = 0.0;
+  };
+  inline void clearFT() {
+    ft.clear();
+    ft_time = 0;
+
+    ftMAX.fe = 0;
+    ftMAX.te = 0;
+  };
+  inline void clearTA() {
+    for (int i = 0; i < 6; i++)
+      ta[i].clear();
+    ta_time = 0;
+  };
+
+  QList<DI_Transfer> rt;
+  QList<DI_Transfer> wt;
+  QList<DI_Errc> errc;
+  QList<DI_JB> jb;
+  QList<DI_FT> ft;
+  QList<DI_TA> ta[6];
+
+  TOT_Errc errcTOT;
+  DI_Errc errcMAX;
+  AVG_Errc errcAVG;
+  MM_JB jbMM;
+  DI_FT ftMAX;
+
+  float rt_time;
+  float wt_time;
+  float errc_time;
+  float jb_time;
+  float ft_time;
+  float ta_time;
+};
+
+struct MediaInfo {
+  QString type;
+  QString label;
+
+  //	bool		isCD;
+  //	bool		isDVD;
+
+  int spd1X;
+  //	int			itype;
+  QString category;
+  QString mid;
+  QString layers;
+  int ilayers;
+  QString gbpl;
+  int igbpl;
+  QString prot;
+  QString regions;
+
+  int creads;
+  int creadm;
+  QString creadmsf;
+  int cfrees;
+  int cfreem;
+  QString cfreemsf;
+  int ctots;
+  int ctotm;
+  QString ctotmsf;
+
+  QString erasable;
+  QString dstate;
+  QString sstate;
+  QStringList rspeeds;
+  QStringList wspeedsd;
+  QStringList wspeedsm;
+  QStringList tspeeds_errc;
+  int tdata_errc;
+  QStringList tspeeds_jb;
+
+  float grec;
+  QString writer;
+};
+
+struct TestSpeeds {
+  int32_t rt, wt, errc, jb, ft;
+};
+
+class MediaWatcher;
+class ResultsReader;
+class ResultsWriter;
+
+class device : public QObject {
+  Q_OBJECT
+public:
+  enum devtype {
+    DevtypeNone = 0,
+    DevtypeVirtual = 1,
+    DevtypeLocal = 2,
+    DevtypeTCP = 3
+  };
+
+  enum ThreadType {
+    threadNone = 0,
+    threadDevice = 1,
+    threadMedia = 2,
+    threadTest = 3,
+    threadGetFeatures = 4,
+    threadGetASDB = 5,
+    threadMQCK = 6,
+    threadAScre = 7,
+    threadDestruct = 8,
+    threadTattoo = 9
+  };
+
+  device(QObject *p);
```

Comment:
Codacy found an issue: [Class 'device' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)